iModelCore/ECPresentation/Source/Content/ContentDescriptorBuilder.cpp

```diff
@@ -1271,12 +1310,19 @@ struct ContentDescriptorBuilderImpl : ContentSpecificationsHandler
     void _AppendClass(SelectClassInfo const& classInfo) override
         {
         m_descriptor->AddSelectClass(classInfo, m_specification ? m_specification->GetHash() : "");
+        }
 
+    /*---------------------------------------------------------------------------------**//**
+    * @bsimethod
+    +---------------+---------------+---------------+---------------+---------------+------*/
+    PropertyAppendResult _OnPropertiesAppended(PropertyAppender& appender, ECClassCR propertyClass, Utf8StringCR classAlias, bool includeNested) override

```

Comment:
`includeNested: false` - ignore `applyOnNestedContent` attribute and add calculatedProperties from content modifiers.
`includeNested: true` - check for `applyOnNestedContent` to be true and just then add calculatedProperties. I also don't like this solution. Made a differrent approach by checking what kind of appender we have
![image](https://user-images.githubusercontent.com/75395229/228301273-bdda05d2-f394-47b1-aab2-b797d6dfeae5.png)
Not sure if this is a good idea

Reply:
I think the approach to use appender to get this information is great. Maybe, instead of using `dynamic_cast`, you could instead use a virtual function.